sdk/nodejs/appservice/zMixins.ts

```diff
@@ -400,6 +400,46 @@ export interface Function {
     appSettings?: pulumi.Input<{ [key: string]: any }>;
 }
 
+// Type guard for Function interface
+function isFunction(object: any): object is Function {
+    return 'name' in object && 'bindings' in object && 'callback' in object;

```

Comment:
i prefer the following:

`return obj && (<Function>obj).name && ...`

this way if we change anything about the shape of Function, we get an error here.  Right now, this is purely dynamic, so TS can't check anything.

Reply:
Great, thanks for the hint!